feature_engine/dataframe_checks.py

```diff
@@ -85,6 +86,108 @@ def check_X(X: Union[np.generic, np.ndarray, pd.DataFrame]) -> pd.DataFrame:
     return X
 
 
+def check_y(
+        y: Union[np.generic, np.ndarray, pd.Series, List],
+        multi_output: bool = False,
+        y_numeric: bool = True,
+) -> pd.Series:
+    """
+    Checks that y is a series, or alternatively, if it can be converted to a series.
+
+    Parameters
+    ----------
+    y : pd.Series, np.array, list
+
+    multi_output : bool, default=False
+        Whether to allow 2D y (array). If false, y will be
+        validated as a vector. y cannot have np.nan or np.inf values if
+        multi_output=True.
+
+    y_numeric : bool, default=False
+        Whether to ensure that y has a numeric type. If dtype of y is object,
+        it is converted to float64. Should only be used for regression
+        algorithms.
+
+    Returns
+    -------
+    y: pd.Series
+    """
+
+    if y is None:
+        raise ValueError("y cannot be None.")
+
+    elif isinstance(y, pd.Series):
+        _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
```

Comment:
by not assigning, you get all the checks without the conversion, but e.g. the numeric conversion also doesn't happen...

Reply:
Good catch, thank you! Refactoring.